jsonnet/components/cluster-monitoring-operator.libsonnet

```
@@ -345,6 +347,43 @@ function(params) {
     }],
   },
 
+  // This role provides read access to the user-workload Alertmanager API.
+  // We use a fake subresource 'api' to map to the /api/* endpoints of the
+  // Alertmanager API.
+  // Using "nonResourceURLs" doesn't work because authenticated users and
+  // service accounts are allowed to get /api/* by default.
+  userWorkloadAlertmanagerApiReader: {
+    apiVersion: 'rbac.authorization.k8s.io/v1',
+    kind: 'Role',
+    metadata: {
+      name: 'monitoring-alertmanager-api-reader',
+      namespace: cfg.namespaceUserWorkload,
+    },
+    rules: [{
+      apiGroups: ['monitoring.coreos.com'],
+      resources: ['alertmanagers'],
```

shouldn't it target the api subresource?
```suggestion
      resources: ['alertmanagers/api'],
```

---

done